setting/#26/migrate to Yarn Berry with PnP

yihimin opened the pull request:
## 📌 PR 제목

[chore] #26 - Yarn Berry + PnP 마이그레이션 적용

## ✨ 변경 사항

- 레거시 lockfile 및 node_modules 제거 (`package-lock.json`, `yarn.lock`, `node_modules/`)
- Yarn v2 (Berry)로 버전 업 및 `.yarnrc.yml` 설정 추가 (`nodeLinker: pnp`)
- `.gitignore`에 `.yarn/*`, `.pnp.*` 관련 설정 추가
- `yarn install`로 의존성 재설치 및 PnP 적용 완료

## 🧐 PR 체크리스트

- [x] 코딩 스타일을 따랐나요?
- [x] 관련 이슈를 연결했나요?
- [x] 리뷰어가 이해할 수 있도록 설명을 작성했나요?

## 📷 스크린샷 (선택 사항)
![image](https://github.com/user-attachments/assets/e2021814-d615-47fb-b369-5c7325bb8a9f)

![image](https://github.com/user-attachments/assets/e0b19766-49b2-4d49-9494-a346b367553f)

## 🚀 관련 이슈

Closes #26

## 🧩 리뷰어에게 바라는 점

- 현재 PnP 환경에서 한글(`metadata.title`, `description`) 사용 시 `ByteString` 관련 에러가 발생합니다.
  → PnP 방식을 유지한 채로 해결 가능한지, 아니면 `nodeLinker: node-modules` 방식으로 전환하는 게 나을지 의견 부탁드립니다.

yihimin committed:
chore: migrate to Yarn Berry with PnP

soonki-98 commented on .yarnrc.yml:
[요런 글](https://medium.com/wantedjobs/yarn-classic%EC%97%90%EC%84%9C-pnpm%EC%9C%BC%EB%A1%9C-%EC%A0%84%ED%99%98%ED%95%98%EA%B8%B0-with-turborepo-7c0c37cb3f9e)에서는 .npmrc에서 설정했길래 여쭤봤습니다

soonki-98 reviewed (dismissed):
pnp만 보고 pnpm인줄 알았는데 아니었군요... 부끄럽네요

yihimin committed:
feat: update project metadata and add ESLint/TypeScript SDKs

- Changed project title and description in metadata.ts for clarity.
- Added ESLint SDK files including bin, lib, and types for better integration.
- Introduced TypeScript SDK files with necessary configurations and scripts.
- Updated package.json for both ESLint and TypeScript to reflect new versions and structure.
- Generated integrations.yml for VSCode support.
